src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java

```diff
@@ -214,6 +218,38 @@ public String getValue(String configKey) {
                 Optional<BrokerConfig> saslBroker = root.kafka.brokers.stream()
                         .filter(broker -> "sasl".equals(broker.authtype))
                         .findAny();
+                Optional<BrokerConfig> sslBroker = root.kafka.brokers.stream()
+                    .filter(broker -> "SSL".equals(broker.securityProtocol))
+                    .findAny();
+
+                if (saslBroker.isPresent() || sslBroker.isPresent()) {
+                    switch (configKey) {
+                        case KAFKA_SECURITY_PROTOCOL_KEY:
+                        case CAMEL_KAFKA_SECURITY_PROTOCOL_KEY:
+                            if (saslBroker.isPresent()) {
+                                return saslBroker.get().sasl.securityProtocol;
+                            } else {
+                                return sslBroker.get().securityProtocol;
+                            }
+                        case KAFKA_SSL_TRUSTSTORE_LOCATION_KEY:
+                        case CAMEL_KAFKA_SSL_TRUSTSTORE_LOCATION_KEY:
+                            if (saslBroker.isPresent()) {
+                                return createTempKafkaCertFile(saslBroker.get().cacert);
+                            } else {
+                                return createTempKafkaCertFile(sslBroker.get().cacert);
+                            }
+                        case KAFKA_SSL_TRUSTSTORE_TYPE_KEY:
+                        case CAMEL_KAFKA_SSL_TRUSTSTORE_TYPE_KEY:
+                            if(null != getValue(KAFKA_SSL_TRUSTSTORE_LOCATION_KEY)) {
+                                return KAFKA_SSL_TRUSTSTORE_TYPE_VALUE;
+                            }
+                        default:
+                            if (sslBroker.isPresent()) {
```

Comment:
I understand why this condition is needed, but it's a bit confusing from a maintenance perspective.

Could we have a clear separation between SASL and SSL?
```java
if (saslBroker.isPresent()) {
    // old switch, unchanged
} else {
    Optional<BrokerConfig> sslBroker = root.kafka.brokers.stream()
        .filter(broker -> "SSL".equals(broker.securityProtocol))
        .findAny();
    if (sslBroker.isPresent()) {
        // new switch, dedicated to SSL only
    }
}
```